### PR TITLE
Release checklist update: publishing solc-js tarball from CI

### PR DESCRIPTION
**Pull request by cameel**
Depends on https://github.com/ethereum/solc-js/pull/717. This will only be possible when we merge that PR in solc-js.

### CODE REVIEW

File: ReleaseChecklist.md [hunk 1]
```diff
@@ -94,9 +95,11 @@ At least a day before the release:
 ### Release solc-js
  - [ ] Wait until solc-bin was properly deployed. You can test this via remix - a test run through remix is advisable anyway.
  - [ ] Increment the version number, create a pull request for that, merge it after tests succeeded.
- - [ ] Run ``npm run build:tarball`` in the updated ``solc-js`` repository to create ``solc-<version>.tgz``. Inspect the tarball to ensure that it contains an up to date compiler binary.
- - [ ] Run ``npm run publish:tarball`` to publish the newly created tarball.
  - [ ] Create a tag using ``git tag --annotate v$VERSION`` and push it with ``git push --tags``.
+ - [ ] Wait for the CI runs on the tag itself.
+ - [ ] Take the ``solc-x.y.z.tgz`` tarball from ``build-package`` run of the tagged commit in circle-ci.
```
Comment (r0qs) [inline ReleaseChecklist.md]: I would add `artifact` just to make clear where the tarball should be located in the circleci job dashboard.

```suggestion
 - [ ] Take the ``solc-x.y.z.tgz`` tarball artifact from ``build-package`` run of the tagged commit in circle-ci.
```

Comment (cameel) [reply ReleaseChecklist.md]: Done.

**Review by r0qs — APPROVED** [review]
LGTM. Just need a rebase.

**Comment by ekpyron** [comment]
FYI, I'm fine with merging this if @r0qs and @cameel agree that it's good :-). We just need to be extra careful in double-checking what comes out of this in the next release.

**Comment by cameel** [comment]
Just note that https://github.com/ethereum/solc-js/pull/717 is not merged yet due to breakage caused by Hardhat's update to pnpm. Which should be fixed by https://github.com/ethereum/solc-js/pull/719, but we need to push both of these through before we can merge this checklist update.

**Comment by cameel** [comment]
Well, I just reviewed @r0qs fix and it's great and my PR also rebased on it without issues. So I merged it all and now we can also merge this.

**Comment by christianlantin** [comment]
Mastet

**Comment by christianlantin** [comment]
Master

**Pull request merged**